mypy/semanal.py

```diff
@@ -1335,11 +1335,11 @@ def visit_import_from(self, imp: ImportFrom) -> None:
                     self.add_submodules_to_parent_modules(possible_module_id, True)
                 elif possible_module_id in self.missing_modules:
                     missing = True
-            # If it is still not resolved, and the module is a stub
-            # check for a module level __getattr__
-            if module and not node and module.is_stub and '__getattr__' in module.names:
+            # If it is still not resolved, check for a module level __getattr__
+            if (module and not node and (module.is_stub or self.options.python_version >= (3, 7))
```

Comment:
I think that's different actually. In `checker.py` we don't show an error if the user uses module-level `__getattr__` before 3.7, because it's harmless (it does nothing, but also doesn't break anything). Here though we're essentially implementing the equivalent of the runtime attribute lookup on modules, so we should only look at `__getattr__` in 3.7 or later.